Tools/autotest/arducopter.py

```diff
@@ -4719,16 +4745,22 @@ def loiter_requires_position(self):
         # ensure we can't switch to LOITER without position
         self.progress("Ensure we can't enter LOITER without position")
         self.context_push()
-        self.set_parameter("GPS_TYPE", 0)
+        self.set_parameter("GPS_TYPE", 2)
+        self.set_parameter("SIM_GPS_DISABLE", 1)
         self.reboot_sitl()
-        self.change_mode('STABILIZE')
-        self.wait_ekf_flags((mavutil.mavlink.ESTIMATOR_ATTITUDE |
+
+        # check for expected EKF flags
+        ahrs_ekf_type = self.get_parameter("AHRS_EKF_TYPE")
+        expected_ekf_flags = (mavutil.mavlink.ESTIMATOR_ATTITUDE |
                              mavutil.mavlink.ESTIMATOR_VELOCITY_VERT |
                              mavutil.mavlink.ESTIMATOR_POS_VERT_ABS |
-                             mavutil.mavlink.ESTIMATOR_CONST_POS_MODE |
-                             mavutil.mavlink.ESTIMATOR_PRED_POS_HORIZ_REL),
-                            0,
-                            timeout=120)
+                             mavutil.mavlink.ESTIMATOR_CONST_POS_MODE)
+        if ahrs_ekf_type == 2:
+            expected_ekf_flags = expected_ekf_flags | mavutil.mavlink.ESTIMATOR_PRED_POS_HORIZ_REL
```

Comment:
Wow. That's not a small difference between the two estimators!

Reply:
This was lightly discussed during the previous review (see resolved conversation).  I've raised an issue (https://github.com/ArduPilot/ardupilot/issues/16018) but I think EKF2 is incorrect.